common/http.hpp

```diff
@@ -0,0 +1,220 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_COMMON_HTTP_HPP
+#define MENDER_COMMON_HTTP_HPP
+
+#include <common/error.hpp>
+#include <common/events.hpp>
+#include <common/expected.hpp>
+#include <common/io.hpp>
+#include <common/log.hpp>
+
+#ifdef MENDER_EVENTS_USE_BOOST
+#include <boost/asio.hpp>
+#include <boost/beast.hpp>
+#endif // MENDER_EVENTS_USE_BOOST
+
+#include <functional>
+#include <unordered_map>
+
+namespace mender {
+namespace http {
+
+using namespace std;
+
+#ifdef MENDER_EVENTS_USE_BOOST
+namespace asio = boost::asio;
+namespace beast = boost::beast;
+namespace http = beast::http;
+#endif // MENDER_EVENTS_USE_BOOST
+
+namespace error = mender::common::error;
+namespace events = mender::common::events;
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+namespace log = mender::common::log;
+
+class HttpErrorCategoryClass : public std::error_category {
+public:
+	const char *name() const noexcept override;
+	string message(int code) const override;
+};
+extern const HttpErrorCategoryClass HttpErrorCategory;
+
+enum ErrorCode {
+	NoError = 0,
+	NoSuchHeaderError,
+	InvalidUrlError,
+	BodyMissingError,
+};
+
+error::Error MakeError(ErrorCode code, const string &msg);
+
+enum Verb {
```

Comment:
Why not `Method`?

Reply:
For some reason it seemed natural when I wrote it, but I agree, I'll switch to `Method`.

Reply:
Ah, it was because of Beast, they call it verb. I still preger Method though.